src/meta/src/hummock/hummock_manager.rs

```diff
@@ -260,11 +262,80 @@ where
                     .levels
                     .insert(compaction_group.group_id(), Levels { levels });
             }
-            init_version.insert(self.env.meta_store()).await?;
-            versioning_guard
-                .hummock_versions
-                .insert(init_version.id, init_version);
+            if version_ids.is_empty() {
+                version_ids.insert(init_version.id);
+            }
+            let mut redo_state = init_version.clone();
+            if version_ids.contains(&init_version.id) {
+                init_version.insert(self.env.meta_store()).await?;
+                versioning_guard
+                    .hummock_versions
+                    .insert(init_version.id, init_version);
+            }
+
+            for (id, version_delta) in &hummock_version_deltas {
+                for (compaction_group_id, level_deltas) in &version_delta.level_deltas {
+                    let mut l0_remove_position = None;
```

Comment:
It's difficult to understand. Why no just reply deltas just by `Vec<HashSet<SstTable>>`

Reply:
I means that we can give a function which just like:
```rust
impl XXXX for HummockVersion {
   fn apply_to_version(&mut self, delete_table_ids: Vec<u64>, new_tables: Vec<u64>)  -> Result<()> {
   }
}
```

And then `report_compact_task`/ `commit_epoch`/ `load_meta_store_state` and share this function.

Reply:
> I means that we can give a function which just like:
> 
> ```rust
> impl XXXX for HummockVersion {
>    fn apply(&mut self, delete_table_ids: Vec<u64>, new_tables: Vec<SstableInfo>)  -> Result<()> {
>    }
> }
> ```
> 
> And then `report_compact_task`/ `commit_epoch`/ `load_meta_store_state` and share this function.

fixed